src/Components/Artist/MarketInsights.tsx

```diff
@@ -0,0 +1,133 @@
+import * as React from "react"
+import { createFragmentContainer, graphql } from "react-relay"
+
+export interface MarketInsightsProps extends RelayProps, React.HTMLProps<MarketInsights> {}
+
+const Categories = {
+  "blue-chip": "Blue Chip",
+  "top-established": "Top Established",
+  "top-emerging": "Top Emerging",
+}
+
+export class MarketInsights extends React.Component<MarketInsightsProps, null> {
+  renderGalleryRepresentation() {
+    const { highlights } = this.props.artist
+    const { partners } = highlights
+    if (partners && partners.edges && partners.edges.length > 0) {
+      const partner = partners.edges[0].node
+      const { categories } = partner
+      let category
+      Object.keys(Categories).forEach(key => {
+        categories.forEach(partnerCategory => {
+          if (partnerCategory.id === key) {
+            category = Categories[key]
+          }
+        })
+      })
+      return (
+        <div>
+          Represented by {partner.name}
+          <br />
+          This is a {category} type of partner.
+        </div>
+      )
+    }
+  }
+
+  renderAuctionHighlight() {
+    if (!this.props.artist.auctionResults || this.props.artist.auctionResults.edges.length < 1) {
+      return null
+    }
+    const topAuctionResult = this.props.artist.auctionResults.edges[0].node
+
+    return (
+      <div>
+        Top auction result is {topAuctionResult.price_realized} at {topAuctionResult.organization}
+      </div>
+    )
+  }
+
+  renderPermanentCollection() {
+    const { collections } = this.props.artist
+    if (collections && collections.length > 0) {
+      return (
+        <div>
+          Collected by major museums
+          <br />
+          {collections.join(", ")}
+        </div>
+      )
+    }
+  }
+
+  render() {
+    return (
+      <div>
+        {this.renderGalleryRepresentation()}
+        <br />
+        {this.renderAuctionHighlight()}
+        <br />
+        {this.renderPermanentCollection()}
+      </div>
+    )
+  }
+}
+
+export default createFragmentContainer(
+  MarketInsights,
+  graphql.experimental`
+    fragment MarketInsights_artist on Artist {
+      _id
+      collections
+      highlights {
+        partners(first: 1, represented_by: true, partner_category: ["blue-chip", "top-established", "top-emerging"]) {
```

Comment:
Is there a way to reuse the list above rather than re-hard-code it here? (I realize these are different "languages," but maybe there's a pattern for this?)

Reply:
Are you referring to the TS typings below? If so, those should simply be emitted by the compiler based on the selections made in the fragment, but we had to merge this branch first before we could continue on expanding the compiler to automatically emit TS typings (the only current available upstream version emits Flow typings).

Reply:
I _think_ @joeyAghion meant the list of category slugs. I don't believe you can do string interpolation within this GraphQL string, but you can definitely make these arguments proper params and pass variables in.

That would mean I'd add the `@argumentDefinitions` keyword and would have to keep `graphql.experimental` (which is fine).

Reply:
Ah I see 👍